src/hooks/useOnboardingMessages/index.ts

```diff
@@ -0,0 +1,10 @@
+import {useMemo} from 'react';
+import useLocalize from '@hooks/useLocalize';
+import getOnboardingMessages from '@libs/actions/Welcome/OnboardingFlowTasks';
+
+export default function useOnboardingMessages() {
+    const {translate} = useLocalize();
+    // eslint-disable-next-line react-compiler/react-compiler, react-hooks/exhaustive-deps
+    const onboardingMessages = useMemo(() => getOnboardingMessages(), [translate]);
+    return onboardingMessages;
```

Comment:
```suggestion
    const {preferredLocale} = useLocalize();

    // Adding preferredLocale as dependency because getOnboardingMessages depends on the current locale
    // eslint-disable-next-line react-compiler/react-compiler, react-hooks/exhaustive-deps
    const onboardingMessages = useMemo(() => getOnboardingMessages(), [preferredLocale]);

    return onboardingMessages;
```